diff --git a/components/content_settings/core/browser/content_settings_pref_provider.cc b/components/content_settings/core/browser/content_settings_pref_provider.cc
index a00cacb48f3b91597edc46926bed38f8ed837c35..4c9ea9c7d75e185921b3898a8de033d592187b08 100644
--- a/components/content_settings/core/browser/content_settings_pref_provider.cc
+++ b/components/content_settings/core/browser/content_settings_pref_provider.cc
@@ -198,6 +198,7 @@ bool PrefProvider::SetWebsiteSetting(
   // sites/origins defined by the |primary_pattern| and the |secondary_pattern|.
   // Default settings are handled by the |DefaultProvider|.
   if (primary_pattern == ContentSettingsPattern::Wildcard() &&
+      BRAVE_SET_WEBSITE_SETTINGS
       secondary_pattern == ContentSettingsPattern::Wildcard()) {
     return false;
   }
