diff --git a/components/content_settings/core/browser/content_settings_default_provider.cc b/components/content_settings/core/browser/content_settings_default_provider.cc
index 301e38ca2337c5596a197c6d512a52cf0644cfb2..ce843ab34142af78d4fa5af1978f7719f9ffbabb 100644
--- a/components/content_settings/core/browser/content_settings_default_provider.cc
+++ b/components/content_settings/core/browser/content_settings_default_provider.cc
@@ -284,6 +284,7 @@ std::unique_ptr<RuleIterator> DefaultProvider::GetRuleIterator(
     return nullptr;
 
   base::AutoLock lock(lock_);
+  BRAVE_EARLY_RETURN_ON_SHIELDS_CONTENT_SETTINGS_TYPE
   auto it = default_settings_.find(content_type);
   if (it == default_settings_.end()) {
     NOTREACHED();
@@ -431,6 +432,7 @@ void DefaultProvider::DiscardOrMigrateObsoletePreferences() {
   }
   prefs_->ClearPref(kDeprecatedNativeFileSystemWriteGuardDefaultPref);
 #endif  // !defined(OS_ANDROID)
+BRAVE_DISCARD_OR_MIGRATE_OBSOLETE_PREFERENCES
 }
 
 }  // namespace content_settings
